sherpa/astro/plot.py

```diff
@@ -64,25 +64,43 @@ def _check_hist_bins(plot):
 
     Parameters
     ----------
-    plot
-        The plot structure, which must have xlo and xhi attributes.
+    xlo, xhi : array
+        Lower and upper bin boundaries. Typically, ``xlo`` will contain the
+        lower boundary and ``xhi`` the upper boundary, but this function can
+        deal with situations where that is reversed. Both arrays have to be
+        monotonically increasing or decreasing.
+
+    Returns
+    -------
+    xlo, xhi : array
+        xlo and xhi with values that were very close (within numerical
+        tolerance) before changed such that they now match exactly.
 
     Notes
     -----
     Note that this holds even when plotting wavelength values, who
     have xlo/xhi in decreasing order, since the lo/hi values still
     hold.
-
     """
-
     # Technically idx should be 0 or 1, with no -1 values. We
     # do not enforce this. What we do is to take all bins that
     # appear similar (sao_fcmp==0) and set the xlo[i+1] bin
     # to the xhi[i] value.
     #
-    equal = sao_fcmp(plot.xlo[1:], plot.xhi[:-1], _tol)
+    # Deal with xhi <-> xlo switches. Those can occor when converting
+    # from energy to wavelength.
+    # Deal with reversed order. Can happen when converting from energy
+    # to wavelength, or if input PHA is not ordered in increasing energy.
+    # But is both are happening at the same time, need to switch twice, which
+    # is a no-op. So, we get to use the elusive Python XOR operator.
+    if (xlo[0] > xhi[0]) ^ (xhi[0] > xhi[-1]):
+        xlo, xhi = xhi, xlo
+    equal = sao_fcmp(xlo[1:], xhi[:-1], _tol)
+    print(equal)
```

Comment:
We should remove this debug line

Reply:
Maybe you need to reload? I don't see that print line and github mark the view of this comment as "outdated"

Reply:
I'm viewing by commits rather than the whole commit. I see it's removed in the next commit.